pkg/router/appmesh.go

```diff
@@ -187,9 +188,11 @@ func (ar *AppMeshRouter) reconcileVirtualNode(canary *flaggerv1.Canary, name str
 
 	// update virtual node
 	if virtualnode != nil {
-		if diff := cmp.Diff(vnSpec, virtualnode.Spec); diff != "" {
+		if diff := cmp.Diff(vnSpec, virtualnode.Spec); diff != "" || compareObjMeta(newMetadata, virtualnode) {
```

Comment:
sorry i didn't catch this earlier, but i think it'd be a better idea to use `cmp.Diff` instead of our own util func, as it'd increase maintenance burden and we are already using it everywhere else to compare stuff.
so we can have something like :
```suggestion
		specDiff := cmp.Diff(vnSpec, virtualnode.Spec)
		labelsDiff := cmp.Diff(newMetadata.Labels, virtualNode.Metadata.Labels)
		annotationsDiff := cmp.Diff(newMetadata.Annotations, virtualNode.Metadata.Annotations)
	     if specDiff != "" || labelsDiff != "" || annotationsDiff != ""{
```